src/app/lib/analyticsUtils/chartbeat.test.js

```diff
@@ -0,0 +1,158 @@
+import {
+  chartbeatUID,
+  useCanonical,
+  getSylphidCookie,
+  getDomain,
+  buildSections,
+  getType,
+} from './chartbeat';
+import onClient from '../utilities/onClient';
+
+let isOnClient = false;
+
+jest.mock('../utilities/onClient', () => jest.fn());
+onClient.mockImplementation(() => isOnClient);
+
+describe('Chartbeat utilities', () => {
+  it('should return the correct chartbeat UID', () => {
+    expect(chartbeatUID).toBe(50924);
+  });
+
+  it('useCanonical should be true', () => {
+    expect(useCanonical).toBe(true);
+  });
+
+  describe('Chartbeat ID Cookie', () => {
+    it('should return null when onClient is false', () => {
+      expect(getSylphidCookie()).toBe(null);
+    });
+
+    it('should return null when ID cookie does not exist', () => {
+      isOnClient = true;
+      expect(getSylphidCookie()).toBe(null);
+    });
+
+    it('should return the contents of the ID cookie when a value is present', () => {
+      const expectedCookieValue = 'foobar';
+      isOnClient = true;
+
+      Object.defineProperty(window.document, 'cookie', {
+        writable: true,
+        value: `ckns_sylphid=${expectedCookieValue}`,
+      });
+
+      expect(getSylphidCookie()).toBe(expectedCookieValue);
```

Comment:
We're testing `js-cookie` behaviour here rather than our own logic, no?

Reply:
This series of tests test the following:

- `onClient` is `false` (the first branch of the function under test)
- `onClient` is `true`, and `ckns_sylphid` returns a truthy value (the second branch of the function under test)
- `onClient` is `true`, but `ckns_sylphid` returns a falsy value (the third branch of the function under test)

It does this by mocking the response of `document.cookie`, so yes, it's testing both the `getSylphidCookie` method, but also `js-cookie`, which probably is unnecessary looking back on it, but shouldn't cause us any issues?

Happy to change this to mock the response of `Cookie.get` if you'd prefer 🙂

Reply:
Thanks for the summary. Yeah, I guess I would just have mocked `Cookie.get` but it works and can't see it causing any issues, happy to leave as-is.

Reply:
Yeah so would have I looking back on it... not sure why I didn't do it that way to begin with. I've changed the test to reflect this.